### PR TITLE
[DebugInfo] Add default true return for unexpected types in isUnsignedType

### PR DESCRIPTION
**Pull request by augusto2112**

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci test llvm

File: llvm/lib/CodeGen/AsmPrinter/DebugHandlerBase.cpp [hunk 1]
```diff
@@ -215,22 +215,26 @@ bool DebugHandlerBase::isUnsignedDIType(const DIType *Ty) {
...
+             Ty->getName() == "decltype(nullptr)")) &&
+           "Unsupported encoding");
+    return Encoding == dwarf::DW_ATE_unsigned ||
+           Encoding == dwarf::DW_ATE_unsigned_char ||
+           Encoding == dwarf::DW_ATE_UTF || Encoding == dwarf::DW_ATE_boolean ||
+           Ty->getTag() == dwarf::DW_TAG_unspecified_type;
+  }
+  // FIXME: the signedness should come from the expression where the type is
+  // used in, not the type itself.
+  return true;
```
Comment (adrian-prantl) [inline llvm/lib/CodeGen/AsmPrinter/DebugHandlerBase.cpp:237]: On which code path dis this originally crash? Just checking whether this still addresses the original issue you wanted to solve.

Comment (augusto2112) [reply llvm/lib/CodeGen/AsmPrinter/DebugHandlerBase.cpp:237]: This was originally a DICompositeType with an `DW_TAG_enumeration_type` tag, whose base type was another DICompositeType.

**Comment by augusto2112** [comment]
@swift-ci test

**Comment by augusto2112** [comment]
@swift-ci test windows

**Comment by augusto2112** [comment]
@swift-ci test llvm

**Review by adrian-prantl — APPROVED** [review]

**Comment by augusto2112** [comment]
@swift-ci test llvm

**Comment by augusto2112** [comment]
@adrian-prantl could you merge this?

I ran the clang test suite locally and am getting the save clang scan deps failure without the patch, so it's unrelated to this change.

**Pull request merged**